roles/slurm/tasks/client.yml

```diff
@@ -158,6 +158,17 @@
     - 'restart_slurmd'
   become: true
 
+- name: 'Install Slurm GRES config file.'
+  ansible.builtin.template:
+    src: "templates/gres.conf"
+    dest: '/etc/slurm/gres.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
+  when: slurm_cluster_gpus_total | default(0) | int > 0
```

Comment:
This means that when all GPUs are removed, the existing gres.conf file will not be updated nor removed. Would it not be better to remove this `when`, so that the file will get updated to only the comments from the template. Or will an "empty" `gres.conf` cause other problems?

Reply:
The slurm.conf line 94 controls the activation and deactivation of GRES (and by this also `gres.conf`).
There the `gres` is activated only when GPU's are present. As the removal of the unused file is not an issue, but more of a "nice to have it clean and tidy" I didn't spend extra time on this.